zingo-memo/src/lib.rs

```diff
@@ -11,159 +11,44 @@ use zcash_encoding::{CompactSize, Vector};
 #[non_exhaustive]
 #[derive(Debug)]
 pub enum ParsedMemo {
-    Version0 { uas: Vec<UnifiedAddress> },
-}
-
-/// Packs a list of UAs into a memo. The UA only memo is version 0 of the protocol
-/// Note that a UA's raw representation is 1 byte for length, +21 for a T-receiver,
-/// +44 for a Sapling receiver, and +44 for an Orchard receiver. This totals a maximum
-/// of 110 bytes per UA, and attempting to write more than 510 bytes will cause an error.
-pub fn create_wallet_internal_memo_version_0(uas: &[UnifiedAddress]) -> io::Result<[u8; 511]> {
-    let mut uas_bytes_vec = Vec::new();
-    CompactSize::write(&mut uas_bytes_vec, 0usize)?;
-    Vector::write(&mut uas_bytes_vec, uas, |mut w, ua| {
-        write_unified_address_to_raw_encoding(&ua, &mut w)
-    })?;
-    let mut uas_bytes = [0u8; 511];
-    if uas_bytes_vec.len() > 511 {
-        Err(io::Error::new(
-            io::ErrorKind::InvalidData,
-            "Too many uas to fit in memo field",
-        ))
-    } else {
-        uas_bytes[..uas_bytes_vec.len()].copy_from_slice(uas_bytes_vec.as_slice());
-        Ok(uas_bytes)
-    }
+    Version0 {
+        uas: Vec<UnifiedAddress>,
+    },
+    Version1 {
+        uas: Vec<UnifiedAddress>,
+        transaction_heights_and_indexes: Vec<(BlockHeight, usize)>,
```

Comment:
Is there a case for a format that doesn't include a `ua` vector?    Since we know that we're send to an internal address?

Reply:
The ua vector is in there for the same reason it already was, marking the full UA we sent to, as we're piggybacking off of change memos at the moment. That Ua vector can also always be empty if it needs to be, so adding a new format version to occasionally save a byte doesn't seem very important even if we start doing these memos in note cultivation.